alea/include/alps/alea.hpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 1998-2017 ALPS Collaboration. See COPYRIGHT.TXT
+ * All rights reserved. Use is subject to license terms. See LICENSE.TXT
+ * For use in publications, see ACKNOWLEDGE.TXT
+ */
+#pragma once
+
+/**
+ * @namespace alps::alea
+ * @brief Set of accumulators and statistical pre-/post-processing operations.
+ *
+ * Overview
+ * --------
+ * TODO
+ *
+ * Accumulators and results
+ * ------------------------
+ * Most accumulators (`mean_acc`) have a matching result class (`mean_result`).
+ * To obtain a result from an accumulator, the accumulators provide both a
+ * `result()` and a `finalize()` method, where
```

Comment:
The problem with mergin `result()` and `finalize()` into one method is const-correctness:

 * `result()` just copies the accumulator data and thus leaves it unchanged: it should therefore be `const` because it would be very surprising to not be able to get a result from an accumulator

 * `finalize()` reaps the data away from the accumulator and thus must be non-`const`.

Merging these methods is strange somehow, since it would either imply different signatures for const and non-const:
```C++
 mean_result result() const;
 mean_result result(bool finalize = false);
```
which is strange since one then cannot do `((const mean_acc<double>)r).result(false);` even though I don't make a copy, or we have the same signature, but rely on a runtime check at const, which is strange too.